### PR TITLE
7 pricing section 21.03.2023

### PR DESCRIPTION
**Pull request by InnaLazarenko**

### CODE REVIEW

File: components/common/buttons/ButtonClose.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import type { FC } from 'react';
+
+import CloseIcon from '@mui/icons-material/Close';
+import IconButton, { type IconButtonProps } from '@mui/material/IconButton';
+
+const ButtonCloseComponent: FC<IconButtonProps> = props => {
+  const { 'aria-label': ariaLabel, size, ...restProps } = props;
+
+  return (
+    <IconButton {...restProps} aria-label="delete">
```
Comment (var-bin) [inline components/common/buttons/ButtonClose.tsx:10]: It looks like `aria-label="delete"` is not the right option. For `IconButton`, it should be something like "Close button". For `CloseIcon`, it should be something like "Close icon".

A better option here will be introducing an id variable and adding it to the label value. The main reason to do so is that `ButtonCloseComponent` can be used across the app in many places. If we use a static value for `aria-label,` all buttons will have the same label. As a result, problems with **a11y**.

File: components/common/buttons/ButtonClose.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      <CloseIcon aria-label={ariaLabel || 'close'} fontSize={size || 'medium'} />
```
Comment (var-bin) [inline components/common/buttons/ButtonClose.tsx:11]: It's unclear why `ariaLabel` passed to `ButtonCloseComponent` is used for `CloseIcon`. From a developer's perspective, I expect that that prop will be used for `IconButton`.

File: components/dialogs/custom-services/CustomService.state.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+const reducer: Reducer<InitialState, ActionChange | ActionBlur> = (state, action) => {
+  switch (action.type) {
+    case 'blur': {
+      const { field } = action;
+
+      return {
+        ...state,
+        [field.name]: {
+          ...state[field.name],
+          success: field.name === 'email' ? !!field.value && field.value.includes('@') : !!field.value,
```
Comment (var-bin) [inline components/dialogs/custom-services/CustomService.state.ts:59]: It looks like `success` state for `'email'` could be moved to a separate function. This action will help us to remove duplication and avoid [the WET anti-pattern](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself#WET).

File: components/pricing/CustomServices.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import Stack from '@mui/material/Stack';
+
+import type { EmptyFunction } from '../../models/common';
+
+import Button from '../common/buttons/Button';
+
+import { CustomServicesContainer } from './styled-components';
+
+interface CustomServicesProps {
+  onOpenDialog?: EmptyFunction;
```
Comment (var-bin) [inline components/pricing/CustomServices.tsx:12]: In my opinion `onOpenDialog` can't be optional

**Review by var-bin — CHANGES_REQUESTED** [review]

File: components/common/buttons/ButtonClose.tsx (hunk 2 above)
Comment (InnaLazarenko) [reply components/common/buttons/ButtonClose.tsx:11]: '+

File: components/common/buttons/ButtonClose.tsx (hunk 1 above)
Comment (InnaLazarenko) [reply components/common/buttons/ButtonClose.tsx:10]: '+

File: components/pricing/CustomServices.tsx (hunk 4 above)
Comment (InnaLazarenko) [reply components/pricing/CustomServices.tsx:12]: '+

**Review by var-bin — APPROVED** [review]

**Pull request merged**